dataset.py

```diff
@@ -119,4 +146,11 @@ def augment(sequence):
         return sequence
 
     # Perform transposition (consider only notes)
-    return (evt + transpose if evt < TIME_OFFSET else evt for evt in sequence)
\ No newline at end of file
+    sequence = (evt + transpose if evt < TIME_OFFSET else evt for evt in sequence)
+
+    # Random time stretch
+    stretch_multiplier = random.uniform(1.0, 2.0)
+
+    # Convert generator to list
+    sequence = list(sequence)
+    return (s for s in stretch_sequence(sequence, stretch_multiplier))
```

Comment:
More optimal: Just return `stretch_sequence(sequence, stretch_multiplier)`

Reply:
augment needs to return a generator

Reply:
If I remember correctly, a list is an iterator/generator.